File: backend/model/game.go
package model

import (
	"time"
)

type Game struct {
	ID          int        `json:"id"`
	ReleaseDate time.Time  `json:"releaseDate"`
	Categories  []Category `json:"categories"`
	Name        string     `json:"name"`
	Description string     `json:"description"`
	Price       float64    `json:"price"`
	PosterUrl   string     `json:"posterUrl"`
}

type GameDetail struct {
	ID                  int        `json:"id"`
	ReleaseDate         time.Time  `json:"releaseDate"`
	Categories          []Category `json:"categories"`
	Name                string     `json:"name"`
	Description         string     `json:"description"`
	Price               float64    `json:"price"`
	PosterUrl           string     `json:"posterUrl"`
	Account             Account    `json:"account"`
	HardwareRequirement []Hardware `json:"hardwareRequirement"`
	Media               []Media    `json:"media"`
}

File: frontend/src/stores/game.ts
import type { GameDetail } from "@/types";
import { defineStore } from "pinia";


export const useGameStore = defineStore('game', {
    state: (): GameDetail => ({
        id: null,
        releaseDate: null,
        categories: null,
        name: '',
        description: '',
        price: 0,
        posterUrl: '',
        account: null,
        hardwareRequirement: null,
        media: null
    }),
    actions: {
        save(gameDetail: GameDetail) {
            this.id = gameDetail.id
            this.releaseDate = gameDetail.releaseDate
            this.categories = gameDetail.categories
            this.name = gameDetail.name
            this.description = gameDetail.description
            this.price = gameDetail.price
            this.posterUrl = gameDetail.posterUrl
            this.account = gameDetail.account
            this.hardwareRequirement = gameDetail.hardwareRequirement
            this.media = gameDetail.media
        },
        reset() {
            this.id = null
            this.releaseDate = null
            this.categories = null
            this.name = ''
            this.description = ''
            this.price = 0
            this.posterUrl = ''
            this.account = null
            this.hardwareRequirement = null
            this.media = null
        }
    }
})